src/internal/connector/exchange/api/mail.go

```diff
@@ -221,6 +225,89 @@ func (c Mail) GetItem(
 	return mail, MailInfo(mail), nil
 }
 
+type mailFolderDeltaPagerer interface {
+	getPage(context.Context) (api.PageLinker, error)
+	setNext(string)
+	valuesIn(api.PageLinker) ([]models.MailFolderable, error)
+}
+
+var _ mailFolderDeltaPagerer = &mailFolderDeltaPager{}
+
+type mailFolderDeltaPager struct {
+	service graph.Servicer
+	builder *users.ItemMailFoldersDeltaRequestBuilder
+}
+
+func NewMailFolderDeltaPager(service graph.Servicer, user string) mailFolderDeltaPagerer {
+	builder := service.Client().
+		UsersById(user).
+		MailFolders().
+		Delta()
+
+	return &mailFolderDeltaPager{service, builder}
+}
+
+func (p *mailFolderDeltaPager) getPage(ctx context.Context) (api.PageLinker, error) {
+	page, err := p.builder.Get(ctx, nil)
+	if err != nil {
+		return nil, graph.Stack(ctx, err)
+	}
+
+	return page, nil
+}
+
+func (p *mailFolderDeltaPager) setNext(nextLink string) {
+	p.builder = users.NewItemMailFoldersDeltaRequestBuilder(nextLink, p.service.Adapter())
+}
+
+func (p *mailFolderDeltaPager) valuesIn(pl api.PageLinker) ([]models.MailFolderable, error) {
+	page, ok := pl.(users.ItemMailFoldersDeltaResponseable)
+	if !ok {
+		return nil, clues.New("unable to convert to ItemMailFoldersDeltaResponseable")
+	}
+
+	return page.GetValue(), nil
+}
+
+var _ mailFolderDeltaPagerer = &mailFolderPager{}
+
+type mailFolderPager struct {
+	service graph.Servicer
+	builder *users.ItemMailFoldersRequestBuilder
+}
+
+func NewMailFolderPager(service graph.Servicer, user string) mailFolderDeltaPagerer {
+	// Sable /mailFolders endpoint does not return any of the nested folders
+	rawURL := fmt.Sprintf(mailFoldersBetaURLTemplate, user)
+	builder := users.NewItemMailFoldersRequestBuilder(rawURL, service.Adapter())
+
+	return &mailFolderPager{service, builder}
+}
+
+func (p *mailFolderPager) getPage(ctx context.Context) (api.PageLinker, error) {
+	page, err := p.builder.Get(ctx, nil)
```

Comment:
should we be passing options like a page size here?

Reply:
The current version does not pass in any options for now. I could add it in a future PR if we should add it.

Reply:
Created https://github.com/alcionai/corso/issues/3243